lms/djangoapps/teams/api.py

```diff
@@ -50,3 +64,110 @@ def discussion_visible_by_user(discussion_id, user):
     """
     team = get_team_by_discussion(discussion_id)
     return not is_team_discussion_private(team) or user_is_a_team_member(user, team)
+
+
+def _is_course_admin(user, course_key):
+    """
+    Returns True if the user is an admin for the course, else returns False
+    """
+    if user.is_staff:
+        return True
+    if CourseStaffRole(course_key).has_user(user) or CourseInstructorRole(course_key).has_user(user):
+        return True
+    return False
+
+
+def has_team_api_access(user, course_key, access_username=None):
+    """Returns True if the user has access to the Team API for the course
+    given by `course_key`. The user must either be enrolled in the course,
+    be course staff, be global staff, or have discussion privileges.
+
+    Args:
+      user (User): The user to check access for.
+      course_key (CourseKey): The key to the course which we are checking access to.
+      access_username (string): If provided, access_username must match user.username for non staff access.
+
+    Returns:
+      bool: True if the user has access, False otherwise.
+    """
+    if _is_course_admin(user, course_key):
+        return True
+    if has_discussion_privileges(user, course_key):
+        return True
+    if not access_username or access_username == user.username:
+        return CourseEnrollment.is_enrolled(user, course_key)
+    return False
```

Comment:
This is the same function copied from views.py, do we want to delete it in view.py?